app/services/mqtt_controller_service.rb

```diff
@@ -47,7 +47,8 @@ def send_to_cloud_controller
     # Send the request
     response = http.request(request)
     Rails.logger.info("Sent message for #{@mqtt_client_guid} #{response.code} #{response.message}")
-    # TODO: We should store the message ID coming back in our task table for tracking
+    @task.update!(:controller_message_id => response.body['id'])
```

Comment:
@hsong-rh We need to check that the status coming back from the Cloud Controller is valid before we can use the body['id']